cmake/DD4hepBuild.cmake

```diff
@@ -71,11 +71,22 @@ macro(dd4hep_set_compiler_flags)
     add_definitions(-DDD4HEP_USE_STDCXX=11)
   endif()
 
+  set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic -Wshadow -Wformat-security -Wno-long-long -Wdeprecated")
```

Comment:
This line is too much. Since it at the top [line 38](https://github.com/AIDASoft/DD4hep/blob/357a977587ae6783be0946223518850b53ac0d42/cmake/DD4hepBuild.cmake#L38) you set the flags you want, then they are checked if the compiler supports them and added to CMAKE_CXX_FLAGS ([line 40-48](https://github.com/AIDASoft/DD4hep/blob/357a977587ae6783be0946223518850b53ac0d42/cmake/DD4hepBuild.cmake#L40))